pkg/cloudprovider/aws/capacity.go

```diff
@@ -33,62 +39,71 @@ func NewCapacity(client ec2iface.EC2API) *Capacity {
 }
 
 // Create a set of nodes given the constraints
-func (cp *Capacity) Create(ctx context.Context, constraints cloudprovider.CapacityConstraints) error {
+func (cp *Capacity) Create(ctx context.Context, constraints *cloudprovider.CapacityConstraints) error {
 	// Convert contraints to the Node types and select the launch template
 	// TODO
 
 	// Create the desired number of instances based on desired capacity
-	config := defaultInstanceConfig("", "", cp.ec2Iface)
-	_ = config
-
-	// Set AvailabilityZone, subnet, capacity, on-demand or spot
-	// and validateAndCreate instances
-	if err := config.validateAndCreate(ctx); err != nil  {
+	// create instances using EC2 fleet API
+	if err := cp.create(ctx, constraints); err != nil {
 		return err
```

Comment:
We might want to think about building annotated errors. May not be necessary here, but something to think about.